core/src/main/java/org/mule/runtime/core/context/notification/Sender.java

```diff
@@ -29,8 +33,12 @@ public void dispatch(ServerNotification notification, NotifierCallback notifier)
         || (null != notification.getResourceIdentifier() && subscriptionFilter.accept(notification.getResourceIdentifier()))) {
       try {
         notifier.notify(pair.getListener(), notification);
-      } catch (Exception e) {
+      } catch (RuntimeException e) {
```

Comment:
Because of 

https://dev.eclipse.org/sonar/coding_rules#rule_key=squid%3AS2221|s=createdAt|asc=false

_Catching Exception seems like an efficient way to handle multiple possible exceptions. Unfortunately, it traps all exception types and so both checked and runtime exceptions, casting too broad a net. Indeed, was it really the intention of developers to also catch runtime exceptions? To prevent any misunderstanding, if both checked and runtime exceptions are really expected to be caught, they should be explicitly listed in the catch clause.
Noncompliant Code Example
try {
  // do something that might throw an UnsupportedDataTypeException or UnsupportedEncodingException
} catch (Exception e) { // Noncompliant
  // log exception ...
}_